internal/stacks/stackruntime/internal/stackeval/component_instance.go

```diff
@@ -1132,6 +1171,39 @@ func (c *ComponentInstance) InspectingState(ctx context.Context) *states.State {
 func (c *ComponentInstance) ResultValue(ctx context.Context, phase EvalPhase) cty.Value {
 	switch phase {
 	case PlanPhase:
+
+		if c.main.PlanningOpts().PlanningMode == plans.DestroyMode {
```

Comment:
Cool, thanks for the extra context.

Since Stacks doesn't really do a graph walk in the same way that Terraform Core does, trying to control when things get evaluated is actually quite tricky. It basically just triggers everything, and then waits for the dependencies to get resolved internally.

Also, Stacks doesn't destroy individual components except by a removed block. This particular check would only be true when the entire stack is being destroyed (ie. every component). 

I think that makes it safe (and easiest) for us just to say that asking for the result of a component during a destroy plan is a non-blocking operation that just returns the prior state values. We then rely on the place we've reversed the dependencies between components to actually enforce the ordering.

In short, I think this is okay 🤞